.NET/Microsoft.Recognizers.Text.DateTime/Extractors/BaseMergedDateTimeExtractor.cs

```diff
@@ -267,7 +267,7 @@ private bool ShouldSkipFromToMerge(ExtractResult er)
 
         private List<ExtractResult> FilterUnspecificDatePeriod(List<ExtractResult> ers)
         {
-            ers.RemoveAll(o => this.config.UnspecificDatePeriodRegex.IsMatch(o.Text));
+            ers.RemoveAll(o => o.Type == "daterange" && this.config.UnspecificDatePeriodRegex.IsMatch(o.Text));
```

Comment:
Can you explain the reason for this? What type of case falls here?

Reply:
We want to filter unspecific date periods like "week", "month" only when they are extracted by DatePeriodExtractor, but not when they are extracted as Duration (otherwise many cases addressed in this PR would also be filtered since we are extracting units when preceded by ordinals, e.g. "first week" -> "week").